.github/workflows/automate-terraform.yml

```diff
@@ -0,0 +1,87 @@
+name: 'Terraform GitHub Actions'
+on:
+  push:
+    branch: [wip_terraform]
+jobs:
+  backend:
+      name: Build SringbootApplication
+      runs-on: ubuntu-latest
+
+      steps:
+      - uses: actions/checkout@v3
+      - name: Set up JDK 11
+        uses: actions/setup-java@v3
+        with:
+          java-version: '17'
+          distribution: 'temurin'
+          cache: maven
+      - name: Build with Maven
+        run: mvn clean install
+      - name: Build & push Docker image
+        uses: mr-smithers-excellent/docker-build-push@v5
+        with:
+          image: yashuyellapu/employee-portal-fullstack
+          tags: backend-latest
+          registry: docker.io
+          dockerfile: Dockerfile
+          username: ${{ secrets.DOCKER_USERNAME }}
+          password: ${{ secrets.DOCKER_PASSWORD }}
+
+  frontend:
+        name: Build ReactJs
+        runs-on: ubuntu-latest
+
+        steps:
+          - uses: actions/checkout@v3
+          - name: Use Node.js
+            uses: actions/setup-node@v3
+            with:
+              node-version: 16
+              cache: 'npm'
+              cache-dependency-path: ./frontend/package-lock.json
+          - name: Install node packages
+            working-directory: ./frontend
+            run: |
+              npm install
+              pwd
+          - name: Build & push Docker image
+            uses: mr-smithers-excellent/docker-build-push@v5
+            with:
+              directory: ./frontend/
+              image: yashuyellapu/employee-portal-fullstack
+              tags: frontend-latest
+              registry: docker.io
+              dockerfile: ./frontend/Dockerfile
+              username: ${{ secrets.DOCKER_USERNAME }}
+              password: ${{ secrets.DOCKER_PASSWORD }}
+  terraform:
+    name: 'Terraform'
+    needs: [backend,frontend]
+    runs-on: ubuntu-latest
+    steps:
+      - name: 'Checkout'
+        uses: actions/checkout@master
+      - name: 'Terraform Init'
+        uses: hashicorp/terraform-github-actions@master
+        with:
+          tf_actions_version: 1.2.1
+          tf_actions_subcommand: 'init'
+          tf_actions_working_dir: '.'
+          tf_actions_comment: true
+          args:
+            '-backend-config="access_key=${{ secrets.AWS_ACCESS_ID }}" -backend-config="secret_key=${{ secrets.AWS_SECRET_KEY}}"'
+        env:
+          aws_access_key: ${{ secrets.AWS_SECRET_KEY }}
+          aws_secret_key: ${{ secrets.AWS_SECRET_KEY }}
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+      - name: 'Deploy terraform'
+        run: terraform apply -auto-approve
```

Comment:
I see that you have used `-auto-approve` which is good and should be used in a CI environment. But are there any problems we need to be careful about?